app/models/session.rb

```diff
@@ -32,6 +42,22 @@ def secure_random_token(length = DEFAULT_TOKEN_LENGTH)
   end
 
   def setup_defaults
-    @token ||= secure_random_token unless persisted?
+    # is this an existing old session without :created_at?
+    session_is_old = @created_at.nil? && persisted?
+
+    @token ||= secure_random_token
+    @created_at ||= Time.now.utc
+
+    # sessions only get saved at creation time.  For an existing session with a nil :created_at,
+    # we must forcibly re-save to redis else :created_at will always be nil and the session could
+    # theoretically last forever.  After being deployed 12 hours, this logic can be deleted then re-deployed
+    save if session_is_old
```

Comment:
Just hit me - I could create a new PR to just add `created_at` to the Session model and deploy 2-21.  Then merge this PR on 2-22 when all Session are guaranteed to have non-nil `created_at` any there will be no need for any migration logic.